nano/core_test/peer_container.cpp

```diff
@@ -11,71 +19,100 @@ TEST (peer_container, empty_peers)
 	ASSERT_EQ (0, network.size ());
 }
 
-TEST (peer_container, no_recontact)
-{
-	nano::test::system system (1);
-	auto & node1 (*system.nodes[0]);
-	nano::network & network (node1.network);
-	auto observed_peer (0);
-	auto observed_disconnect (false);
-	nano::endpoint endpoint1 (boost::asio::ip::address_v6::loopback (), 10000);
-	ASSERT_EQ (0, network.size ());
-	network.channel_observer = [&observed_peer] (std::shared_ptr<nano::transport::channel> const &) { ++observed_peer; };
-	node1.network.disconnect_observer = [&observed_disconnect] () { observed_disconnect = true; };
-	auto channel (network.udp_channels.insert (endpoint1, node1.network_params.network.protocol_version));
-	ASSERT_EQ (1, network.size ());
-	ASSERT_EQ (channel, network.udp_channels.insert (endpoint1, node1.network_params.network.protocol_version));
-	node1.network.cleanup (std::chrono::steady_clock::now () + std::chrono::seconds (5));
-	ASSERT_TRUE (network.empty ());
-	ASSERT_EQ (1, observed_peer);
-	ASSERT_TRUE (observed_disconnect);
-}
-
+// Test a node cannot connect to its own endpoint.
 TEST (peer_container, no_self_incoming)
 {
-	nano::test::system system (1);
-	ASSERT_EQ (nullptr, system.nodes[0]->network.udp_channels.insert (system.nodes[0]->network.endpoint (), 0));
+	nano::test::system system{ 1 };
+	auto & node = *system.nodes[0];
+	node.network.tcp_channels.start_tcp (node.network.endpoint ());
+	auto error = system.poll_until_true (2s, [&node] {
+		auto result = node.network.tcp_channels.find_channel (nano::transport::map_endpoint_to_tcp (node.network.endpoint ()));
+		return result != nullptr;
+	});
+	ASSERT_TRUE (error);
 	ASSERT_TRUE (system.nodes[0]->network.empty ());
 }
 
+// Tests the function nano::transport::tcp_channels.insert () doesn't accept peers from the reserved addresses list
 TEST (peer_container, reserved_peers_no_contact)
```

Comment:
We do not know that the channels are not being inserted for the right reason, it would be better if we added a legal channel and saw the channels number go to 1. that would give us more confidence that the test is testing what it think it is testing and not just failing to insert them an unrelated reason.

Reply:
The test is checking the reserved IP address that cannot be used are not accepted.

Please check the `reserved_address` function it is testing: https://github.com/nanocurrency/nano-node/blob/352f66a2070e22bb9900abd69f016095c58eb983/nano/node/transport/transport.cpp#L114

Reply:
I understand it would be another test.

Reply:
I think it should be in the same test. Because now we don't know that the fails are happening for the correct reason.